Classes/Command/ExportCommand.php

```diff
@@ -63,113 +75,142 @@ protected function execute(InputInterface $input, OutputInterface $output): int
-                                'verify_peer' => false,
-                                'verify_peer_name' => false,
-                                'allow_self_signed' => true,
-                            ],
-                        ]
-                    );
-
-                    $location = (string)$requestUri;
-                    $locationCrumbs = [];
-                    do {
-                        if (strpos($location, '://') === false) {
-                            $location = 'https://' . $row['source_host'] . $location;
-                        }
-                        $actualLocation = $location;
-                        $location = '';
-                        $headers = @get_headers($actualLocation, 1, $context);
-                        if (!$headers) {
-                            break;
-                        }
-                        if (isset($headers['Location'])) {
-                            if (is_array($headers['Location'])) {
-                                $location = array_pop($headers['Location']);
-                            } else {
-                                $location = $headers['Location'];
-                            }
-                        } elseif (isset($headers['location'])) {
-                            if (is_array($headers['location'])) {
-                                $location = array_pop($headers['location']);
-                            } else {
-                                $location = $headers['location'];
-                            }
-                        }
-
-                        if ('' === $location) {
-                            break;
-                        }
-                        if (in_array($location, $locationCrumbs, true)) {
-                            file_put_contents('endless.txt', $location . PHP_EOL, FILE_APPEND);
-                            throw new Exception($row['uid'] . ' seems to endless redirect (' . $location . ')');
-                        }
-                        $locationCrumbs[] = $location;
-                    } while ($location);
-
-                    if (!$locationCrumbs) {
-                        file_put_contents('no_location_header.txt',  $requestUri  . PHP_EOL, FILE_APPEND);
-                        throw new Exception($row['uid'] . ' no location in headers do nothing (' . $requestUri . ')');
-                    }
-
-                    $location = array_pop($locationCrumbs);
-
-                    if ($requestUri->getPath() === $location) {
-                        throw new Exception($row['uid'] . ' source equals target');
-                    }
-                    return $location;
+            $port = $requestUri->getPort();
+            $matchedRedirect = $this->redirectService->matchRedirect(
+                $requestUri->getHost() . ($port ? ':' . $port : ''),
+                $requestUri->getPath(),
+                $requestUri->getQuery() ?? ''
+            );
+
+            if (!$matchedRedirect) {
```

Comment:
@weakbit wann könnte das nicht matchen?

Reply:
Die funktion kann null zurückliefern, darauf sollte man eingehen.
Es kann auch sein dass in zukunft der code so abgeändert wird dass es wichtig wird hier den rückgabewert zu prüfen.